### DISCUSSION TITLE
[Feature]: Limit the use of storage space per account

### DISCUSSION THREAD

**Discussion by DVerdeV** [immich-app/immich#1642] [Feature Request]
### Feature detail

I just found this project looking for a solution to share a storage space with a friend. The thing is that I would like to limit the amount of storage we can use to divide it fairly.

So I propose to create two types of users, a group that is the “users” group and where right now appears the server space, it shows its used and remaining storage. And another group that is the “administrators” group that shows the server space and also their used and remaining storage. 

I think this can help many people who want to share a storage space with several people.

### Platform

Server

**Comment by PixelJonas** [comment]
This feels like an enterprise feature for me. I get your point but for a self-hosted solution shared with friends&family you have a list of users and their used space as an administrator.

If you feel someone is abusing it, you should be able to talk to them about it, rather than enforcing a limit on everybody.

my 2 cent

**Comment by DVerdeV** [comment]
Now that you mention it, I can see the enterprise use, but I said it because in my case I would like to use it with my family, as well as with my friend (we have a fairly large amount of storage). The thing is that we don't want to be on top of it all the time and my family is not very good with technology either, so it would be impossible to instruct them not to go over their storage.

I don't think it would be too complicated to force a limit, as it is calculated in the admin panel.

Sorry I haven't already made a contribution, but I code in Python and Typescript is a bit complicated for me. If I succeed, I'll make a pull request.

**Comment by jrasm91** [comment]
This might be easier to implement/incorporate after the file storage layer is abstracted. (#1011)

It's also worth noting that other layers (os, filesystem, etc.) have some support for these types of features independently of immich.

**Comment by Hobby-Student** [comment]
I would love this feature, too. Personally I wouldn't class this as enterprise feature. People self hosting services like Immich have very different storage capacities. As soon as others are meant to use those services (wife, friends, kids, ...) storage can become the bottleneck. I like to limit each users storage capacity per account, because the "only" problem users can have is that their personal storage limit is running full, but the server itself has some storage left. If some don't have too much storage to not think about it, they can leave the users limit on unlimited.

EDIT
I discovered this project few days ago and currently testing it (web and android). So far it's an awesome project/app.

**Comment by jrasm91** [comment]
We do have the file size in bytes for all the uploaded files. Maybe that would be sufficient for this feature? It would not include generated files (thumbnails, transcoded videos, etc.), so as long as you took that into consideration while allocating space you would be fine. If we do use the data in the database it should hopefully have a minimal perf impact.

**Comment by sypion** [comment]
Would love to see this feature! I host Immich for family and close friends, but adding a quota for each person can help not only encourage them to get rid of/archive photos that aren't needed, but allow me time to possibly migrate their data to another hard drive on the host with more storage. Just an example of a use case.

**Comment by bigtcze** [comment]
I would love to see this feature live. I am currently using Synology Photos and sharing it with dozens of my friends. I really would like to migrate to Immich in the nearest occasion, but I dont want any of my friends use all available storage

**Comment by not-Boris** [comment]
Just was looking for this exact feature, would love to see this go live. Although using immich for family and friends, having storage quotas lets them manage their own library more efficiently, instead of being shown several TB for their allowance. For the non-tech savvy using the service it could get complicated in my experience.

**Comment by cfitzw** [comment]
Dropped in to say that this is a desired feature I'd like to see as well. Setting quotas in the user management screen and then showing them their allocation per their quota is a good self-management indicator. Nextcloud does this and I use it limit family members from consuming all my storage space (before I have to upgrade to the next X TBs).

**Comment by keywal** [comment]
Would appreciate this myself.

**Comment by treverehrfurth** [comment]
I’d love to see this feature as well! Found this discussion after googling if it was possible.

**Comment by samuelemusiani** [comment]
I would be happy if this feature will be implemented!

**Comment by traktuner** [comment]
User quotas would be a nice to have!
Also, at least on my Synology NAS, it shows the full size of the volume to the end user. If I put a quota on the folder where all immich files (loibrary, db, etc) are stored, it still shows the full volume size as available.

**Comment by jrasm91** [comment]
There is a work in progress PR for this feature now.

**Reply by traktuner** [reply]
I know. But the other part of my reply was for the current quota situation where quotas don't display correctly which I hope will also be fixed. I don't want to enforce a quota for every user and they should not see the whole volume size.

**Reply by jrasm91** [reply]
Right, that is why I added a new comment instead of replying to your thread.

**Reply by traktuner** [reply]
lol sorry. got this by email and thought I only get reply-notifications ;)

**Comment by jon-is-me** [comment]
Thanks for working on PR #4471. I don't know if these changes should be added to the existing PR or in a separate follow up PR, but could I also ask for the following to be added:

- Default quota limit for new users
- Quota limit for oidc configured users based on a claim (I'm not too familiar with oidc, so I'm not sure if this request makes sense)

**Reply by jrasm91** [reply]
Yeah, I think we can take a look at both of those. Maybe a storage quota section in the admin settings with an enabled flag, setting for new users, etc.

I can look at a claim mapping for oauth as well.

**Comment by jrasm91** [comment]
Implemented in #4471
